utils.py

```diff
@@ -174,23 +114,87 @@ def annotate_adj(vds):
     return vds.annotate_entries(adj=
                                 (vds.GQ >= adj_gq) & (vds.DP >= adj_dp) & (
                                     ~vds.GT.is_het() |
-                                    ((vds.GT.gtj() == 0) & (vds.AD[vds.GT.gtk()] / vds.DP >= adj_ab)) |
-                                    ((vds.GT.gtj() > 0) & (vds.AD[vds.GT.gtj()] / vds.DP >= adj_ab) &
-                                     (vds.AD[vds.GT.gtk()] / vds.DP >= adj_ab))
+                                    ((vds.GT[0] == 0) & (vds.AD[vds.GT[1]] / vds.DP >= adj_ab)) |
+                                    ((vds.GT[0] > 0) & (vds.AD[vds.GT[0]] / vds.DP >= adj_ab) &
+                                     (vds.AD[vds.GT[1]] / vds.DP >= adj_ab))
                                 )
     )
 
 
-def split_multi_sites(vds):
-    sm = hl.SplitMulti(vds)
-    sm.update_rows(a_index=sm.a_index(), was_split=sm.was_split())
-    return sm.result()
+def add_variant_type(alt_alleles):
+    """
+    Get Struct of variant_type and n_alt_alleles from ArrayExpression of AltAlleles
 
+    :param ArrayExpression alt_alleles: Input ArrayExpression of Strings
+    :return: Struct with variant_type and n_alt_alleles
+    :rtype: Struct
+    """
+    ref = alt_alleles[0]
+    alts = alt_alleles[1:]
+    non_star_alleles = hl.filter(lambda a: a != '*', alts)
+    return Struct(variant_type=
+                  hl.cond(
```

Comment:
I thought about that, but this is a more of a tree, rather than a series of elif's. That is, the first one bifurcates into 2 if else's rather than a series of if, elif, elif, etc.